src/Razor/Microsoft.AspNetCore.Razor.Language/src/TagHelperDescriptorComparer.cs

```diff
@@ -120,8 +120,50 @@ public virtual int GetHashCode(TagHelperDescriptor descriptor)
             hash.Add(descriptor.Kind, StringComparer.Ordinal);
             hash.Add(descriptor.AssemblyName, StringComparer.Ordinal);
             hash.Add(descriptor.Name, StringComparer.Ordinal);
+            hash.Add(descriptor.DisplayName, StringComparer.Ordinal);
+            hash.Add(descriptor.CaseSensitive ? 1 : 0);
+
+            if (descriptor.BoundAttributes != null)
+            {
+                for (var i = 0; i < descriptor.BoundAttributes.Count; i++)
+                {
+                    hash.Add(descriptor.BoundAttributes[i]);
```

Comment:
Would it be better to call `GetHashCode()` on each of the bound attributes? Similiarly for other places. I don't know if it makes a difference. Just wondering.

Reply:
The `HashCodeCombiner.Add` calls `GetHashCode()` underneath (with appropriate null checks and handling).